# MIT License

# Copyright (c) 2020 Hongrui Zheng

# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:

# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.

# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

bridge:
  ros__parameters:
    # topics and namespaces
    ego_scan_topic: 'scan'
    scan_distance_to_base_link: 0.0
    scan_fov: 4.7
    scan_beams: 1080

    # map parameters
    map_path: '/aichallenge/workspace/src/aichallenge_gym_ros/maps/gray_aichallenge'
    map_img_ext: '.png'

    # ego starting pose on map
    sx: 89653.7
    sy: 43122.5
    stheta: 0.0
    # teleop
    kb_teleop: True

    # custom parameters
    mass: 160
    length: 2.0
    width: 1.45
    lf: 5.0435
    lr: 5.0435
    s_max: 0.64
    s_min: -0.64
    a_max: 3.2
    mu: 1.0489
    C_Sf: 5.64718
    C_Sr: 5.65456
    h: 0.2
    I: 81.37
    sv_min: -0.32
    sv_max: 0.32
    v_switch: 7.319
    v_min: -10.0
    v_max: 25.0

    gym_rendering: True